### PR TITLE
updated implementation to use pathlib instead of os.path

### PR DESCRIPTION
**Pull request by erangold**

### CODE REVIEW

File: src/corona_hakab_model/supervisor.py [hunk 1]
```diff
@@ -274,7 +271,7 @@ def get(self, manager: "manager.SimulationManager"):
         pass
 
     def snapshot(self, manager: "manager.SimulationManager"):
-        if manager.current_step % self.interval == 0:
+        if manager.current_step > 0 and manager.current_step % self.interval == 0:
```
Comment (bgmoshe) [inline src/corona_hakab_model/supervisor.py]: Why skip 0? It is the day of the infection. If we want to snapshot it you are disabling this ability

File: src/corona_hakab_model/supervisor.py [hunk 2]
```diff
@@ -47,13 +47,10 @@ def snapshot(self, manager):
             s.snapshot(manager)
 
     def dump(self, filename=None):
-        file_name = Path(filename) if filename else Path(SIM_OUTPUT_FOLDER) / (datetime.now().strftime("%Y%m%d-%H%M%S") + ".csv")
+        output_folder = Path(SIM_OUTPUT_FOLDER)
```
Comment (bgmoshe) [inline src/corona_hakab_model/supervisor.py]: Why do you use Path here? SIM_OUTPUT_FOLDER is supposed to be a path not str

**Review by bgmoshe — COMMENTED** [review]
Small changes

File: src/corona_hakab_model/supervisor.py (hunk 2 above)
Comment (ghost) [reply src/corona_hakab_model/supervisor.py]: SIM_OUTPUT_FOLDER is already a path object, so you can use it as is

File: src/corona_hakab_model/supervisor.py (hunk 1 above)
Comment (erangold) [reply src/corona_hakab_model/supervisor.py]: I thought that 0 won't be so informative, so I skipped it.. I added again in the latest commit

**Review by ghost — APPROVED** [review]

File: src/corona_hakab_model/supervisor.py (hunk 2 above)
Comment (erangold) [reply src/corona_hakab_model/supervisor.py]: Okay, changed it in the latest commit

**Pull request merged**